src/class.c

```diff
@@ -1634,22 +1661,27 @@ bool InterfaceDeclaration::isBaseOf(ClassDeclaration *cd, int *poffset)
     {
         BaseClass *b = cd->interfaces[j];
 
-        //printf("\tbase %s\n", b->base->toChars());
-        if (this == b->base)
+        //printf("\tX base %s\n", b->sym->toChars());
+        if (this == b->sym)
         {
             //printf("\tfound at offset %d\n", b->offset);
             if (poffset)
             {
                 *poffset = b->offset;
-                if (j && cd->isInterfaceDeclaration())
+                // The conversion is interface(cd) to interface(this).
+                if (cd->isInterfaceDeclaration())
```

Comment:
Consider an interface hierarchy similar to COM classes with D interfaces: Whenever you implicitely convert to a base class interface, adding a runtime cast would be awful. 
In contrast, comparing interface pointers for equality happens a lot less often, and excplitely just casting to Object for comparison is faster and explicite. Maybe it could even happen automatically when comparing interface pointers?